### PR TITLE
Fix WASM format link

### PR DESCRIPTION
**Pull request by okyanusoz**
### Description

The spec link in https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format for the binary format was broken. The only alive doc I could find was https://webassembly.github.io/spec/core/bikeshed/#modules%E2%91%A0%E2%93%AA (not exactly sure why the URL is like this but I guess section URLs are like this). This PR replaces the broken link with the correct one.

### Motivation

I was disappointed by the lack of docs available for the binary format.

### Additional details

Found it here: https://webassembly.github.io/spec/ in the W3C version under "5.5.16 Modules".

### Related issues and pull requests

N/A

### CODE REVIEW

File: files/en-us/webassembly/understanding_the_text_format/index.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Let's start with the simplest, shortest possible Wasm module.
 
 This module is totally empty, but is still a valid module.
 
-If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/binary/index.html#high-level-structure):
+If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/bikeshed/#modules%E2%91%A0%E2%93%AA):
```
Comment (Josh-Cena) [inline files/en-us/webassembly/understanding_the_text_format/index.md:34]: The entire `/core/binary/` tree is about binary format; I don't think it's necessary to use another page. What about:

```suggestion
If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/binary/modules.html):
```

Comment (okyanusoz) [reply files/en-us/webassembly/understanding_the_text_format/index.md:34]: Yep that may be better

File: files/en-us/webassembly/understanding_the_text_format/index.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Let's start with the simplest, shortest possible Wasm module.
 
 This module is totally empty, but is still a valid module.
 
-If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/binary/index.html#high-level-structure):
+If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/binary/modules.html):
```
Comment (hamishwillee) [inline files/en-us/webassembly/understanding_the_text_format/index.md:34]: Should we be even more specific in our link?


```suggestion
If we convert our module to binary now (see [Converting WebAssembly text format to Wasm](/en-US/docs/WebAssembly/Text_format_to_Wasm)), we'll see just the 8 byte module header described in the [binary format](https://webassembly.github.io/spec/core/binary/modules.html#binary-module):
```

Comment (okyanusoz) [reply files/en-us/webassembly/understanding_the_text_format/index.md:34]: of course!

**Review by hamishwillee — APPROVED** [review]
Thanks very much!

**Pull request merged**